proxyserver/src/main/java/com/lyft/data/proxyserver/wrapper/MultiReadHttpServletRequest.java

```diff
@@ -52,6 +57,14 @@ public void addHeader(String name, String value) {
     headerMap.put(name, value);
   }
 
+  public void replaceBody(String newBody) throws IOException {
+    this.storeBody(new ByteArrayInputStream(newBody.getBytes()));
+
+    // Also update the content-length
+    Integer contentLength = newBody.getBytes(getCharacterEncoding()).length;
```

Comment:
Hi @puneetjaiswal Added the updated content-length header as you pointed in another pull (I picked up the work from Alex)

https://github.com/lyft/presto-gateway/pull/82

The test is still not passing. the body get replaced fine as long as the text is of the same length. For example, "SELECT 1" gets replaced with "SELECT X" fine. Any idea?

cc @bearcage

Reply:
We may have to externalize the `contentProvider` method in ProxyHandler interface. 
https://github.com/lyft/presto-gateway/pull/82#discussion_r382953957